### PR TITLE
Disable invalid transactions

### PR DESCRIPTION
**Pull request by ryanlabouve**
completes https://app.asana.com/0/home/1201981678382625/1202157876577868

### CODE REVIEW

**Review by mitchtrale — APPROVED** [review]

File: app/components/borrow-flow/borrow.tsx [hunk 1]
```diff
@@ -72,7 +79,7 @@ export default function Borrow({
               ref={inputEl}
               onChange={(e) => setValue(e.target.value)}
               className="bg-transparent text-6xl text-white text-center outline-none"
-              placeholder="0"
+              defaultValue={0}
```
Comment (johnmosesman) [inline app/components/borrow-flow/borrow.tsx:82]: Does this defaultValue help us? It caused weirdness with 0 already being appended to my numbers

File: app/components/borrow-flow/borrow.tsx [hunk 2]
```diff
@@ -1,13 +1,14 @@
 import { ICON_SIZE } from "~/lib/constants";
 import type { SwapRow, SwapRowMarketDatum } from "~/types/global";
-import { useEffect, useState, useRef } from "react";
+import { useEffect, useState, useRef, isValidElement } from "react";
 import type { JsonRpcSigner } from "@ethersproject/providers";
```
Comment (johnmosesman) [inline app/components/borrow-flow/borrow.tsx:4]: warning ^

**Review by johnmosesman — APPROVED** [review]
one warning one question, but lgtm otherwise

File: app/components/borrow-flow/borrow.tsx (hunk 1 above)
Comment (ryanlabouve) [reply app/components/borrow-flow/borrow.tsx:82]: This PR addresses that weirdness

File: app/components/borrow-flow/repay.tsx [hunk 3]
```diff
@@ -38,6 +39,7 @@ export default function Repay({
   let [value, setValue] = useState<string>("");
 
   let inputEl = useRef<HTMLInputElement>(null);
+  let isValid = useValidInput(value, 0, walletBalance);
```
Comment (johnmosesman) [inline app/components/borrow-flow/repay.tsx:42]: Actually you need to have the amount you want to repay in your wallet and you also can't repay more than you've borrowed

**Pull request merged**